chainer/dataset/convert.py

```diff
@@ -139,3 +145,106 @@ def _concat_arrays_with_padding(arrays, padding):
             result[(i,) + slices] = src
 
     return result
+
+
+class ConcatWithAsyncTransfer(object):
+
+    """Interface to concatenate data and transfer them to GPU asynchronously.
+
+    It enables to transfer next batch of input data to GPU while GPU is
+    running kernels for training using current batch of input data.
+    """
+
+    def __init__(self, stream=None):
+        self.stream = stream
+        self.pin_arrays = [[], []]  # numpy arrays with pinned memory
+        self.cp_arrays = [[], []]  # cupy arrays
+
+    def __call__(self, batch, device=None, padding=None):
+        """Concatenate data and transfer them to GPU asynchronously.
+
+        (see concat_examples for detail about concatenation)
+
+        Args:
+            batch (list): A list of examples.
+            device (int): Device ID to which each array is sent.
+            padding: Scalar value for extra elements.
+
+        Returns:
+            Array, a tuple of arrays, or a dictionary of arrays.
+            The type depends on the type of each example in the batch.
+        """
+        if len(batch) == 0:
+            raise ValueError('batch is empty')
+
+        first_elem = batch[0]
+        if (device is None or
+                not isinstance(first_elem, tuple)):
+            # Default concat method is used when device is not set
+            # of input data is not tuple.
```

Comment:
This comment should be moved to the docstring (otherwise just remove the comment because it just explains what one can see in the code).

Reply:
OK, probably will remote the comment.